### PR TITLE
[GNA] fix: embedded export is available for embedded targets only

### PR DESCRIPTION
**Pull request by mkwapul**
### Tickets:
 - 108860

### CODE REVIEW

**Review by sirzabek — COMMENTED** [review]
Do we have any tests for embedded export which could be modified to verify this fix?

**Comment by sirzabek** [comment]
Please use [GNA] in commits so that they can be easily found.

**Comment by mkwapul** [comment]
> Do we have any tests for embedded export which could be modified to verify this fix?

we don't

**Comment by mkwapul** [comment]
> > Do we have any tests for embedded export which could be modified to verify this fix?
> 
> we don't

functional tests (both positive and negative path) added

File: src/plugins/intel_gna/tests/functional/shared_tests_instances/behavior/ov_plugin/core_integration.cpp [hunk 1]
```diff
@@ -341,10 +341,21 @@ TEST(OVClassBasicPropsTest, smoke_SetConfigAfterCreatedLogLevel) {
     ASSERT_THROW(core.set_property("GNA", {{ov::log::level.name(), "NO"}}), ov::Exception);
 }
 
-TEST(OVClassBasicPropsTest, smoke_SetConfigAfterCreatedFwModelPath) {
+TEST(OVClassBasicPropsTest, smoke_SetConfigAfterCreatedFwModelPathNegative) {
```
[the remark this answers is not in the document]
Comment (mkwapul) [reply src/plugins/intel_gna/tests/functional/shared_tests_instances/behavior/ov_plugin/core_integration.cpp:344]: one should throw an exception (negative), while the other not (positive), that's why

[the remark this answers is not in the document]
Comment (mkwapul) [reply src/plugins/intel_gna/tests/functional/shared_tests_instances/behavior/ov_plugin/core_integration.cpp:344]: if insisting on, then please suggest a name

**Review by tadamowicz — APPROVED** [review]

**Pull request merged**